aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -2213,11 +2213,39 @@ std::tuple<Tensor, Tensor> _symeig_helper_cpu(const Tensor& self, bool eigenvect
 }
 
 std::tuple<Tensor, Tensor> symeig(const Tensor& self, bool eigenvectors, bool upper) {
+  TORCH_WARN_ONCE(
+    "torch.symeig is deprecated in favor of torch.linalg.eigh and will be removed in a future ",
+    "PyTorch release.\n",
+    "The default behavior has changed grom using the upper triangular portion of the matrix by default ",
+    "to using the lower triangular portion. This should only matter if the matrix is not symmetric, ",
+    "which is not recommended.\n",
```

Comment:
why is in not recommended? seems like this should just be removed?

Reply:
I certainly think so but numpy-compat...

This function just uses the upper or lower half of the matrix, because it's assumed to be a symmetric matrix, but then the gradien that we return is not upper or lower triangular, but symmetric.This is inconsistent, as a clever user could want to represent two different symmetric matrices with zeros in the diagonal in the same matrix. I this user then used this construction in a differentiable environment, the gradients would be wrong, even though what they were doing seemed semantically right...

In my opinion, we should ask for symmetric matrices, and then user the upper or lower triangular part of the matrix internally, or symmetrise it first to remove possible errors, but imo, these things should not be exposed to the user.

That being said, numpy-compat...

Reply:
I appreciate there's some nuance here and we can dig into it more offline. I just don't like the vagueness of things being not "recommended." Like, either we support things or we don't, and if we don't support them the behavior should be undefined, or defined but subject to change and curse the user who tries to rely on it. 

That said, this is far from the worst thing in our docs today; we can improve it later

Reply:
I forgot to remove these from the comments, I'll do this now.
I also filled an issue describing this problem: https://github.com/pytorch/pytorch/issues/58037